app/src/main/java/com/example/roome/MainActivity.java

```diff
@@ -1,14 +1,80 @@
 package com.example.roome;
 
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.os.Handler;
+
 import androidx.appcompat.app.AppCompatActivity;
 
-import android.os.Bundle;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DatabaseReference;
 
+/**
+ * this activity is presenting the app logo for a few seconds and then disappears.
+ * this activity will be shown on every app visit.
+ */
 public class MainActivity extends AppCompatActivity {
+    public static final String FROM = "called from";
+    public static final String MAIN_SRC = "MAIN";
+
+    private String mUsername;
+    private String mPhotoUrl;
+
+    // Firebase instance variables
+    private FirebaseAuth mFirebaseAuth;
+    private FirebaseUser mFirebaseUser;
+    private DatabaseReference mFirebaseDatabaseReference;
+//    private FirebaseRecyclerAdapter<FriendlyMessage, MessageViewHolder>
+//            mFirebaseAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        final SharedPreferences reader = getApplicationContext().getSharedPreferences(MyPreferences.MY_PREFERENCES, Context.MODE_PRIVATE);
+        final SharedPreferences.Editor editor = reader.edit();
+        editor.putBoolean(MyPreferences.IS_FIRST_LOGIN, true);
+        editor.apply();
+        //Time to launch the next activity
+        int TIME_OUT = 3000;
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                // Initialize Firebase Auth
+                mFirebaseAuth = FirebaseAuth.getInstance();
+                mFirebaseUser = mFirebaseAuth.getCurrentUser();
+                if (mFirebaseUser == null) {
+                    // Not signed in, launch the Sign In activity
+                    startActivity(new Intent(MainActivity.this, SignInActivity.class));
+                    finish();
+                    return;
```

Comment:
why return? shouldn't we keep on running the rest of the function code?it may be his first run or maybe it isn't and he just had to sign in again but then needs to skip to his relevant activity of choosing an apartment/roommate.